framework/tests/viewpointtest/SuperConstructorCalls.java

```diff
@@ -0,0 +1,51 @@
+// Test case for EISOP issue #782:
+// https://github.com/eisop/checker-framework/issues/782
+import viewpointtest.quals.*;
+
+public class SuperConstructorCalls {
+
+    public SuperConstructorCalls() {}
+
+    @SuppressWarnings({"inconsistent.constructor.type", "super.invocation.invalid"})
+    public @ReceiverDependentQual SuperConstructorCalls(@ReceiverDependentQual Object str) {}
+
+    @SuppressWarnings({"inconsistent.constructor.type", "super.invocation.invalid"})
+    public @A SuperConstructorCalls(@A float number) {}
+
+    class Inner extends SuperConstructorCalls {
+        public Inner() {}
+
+        public Inner(Object obj) {
+            super();
+        }
+
+        public Inner(int number, @Top Object obj) {
+            super(obj);
+        }
+
+        // :: warning: (inconsistent.constructor.type)
+        public @A Inner(int number, @A Object obj1, @B Object obj2) {
+            super(obj1);
+        }
+
+        // :: warning: (inconsistent.constructor.type)
+        public @B Inner(int number, @B Object obj1, @A Object obj2, @Top Object obj3) {
+            super(obj1);
```

Comment:
It would be nice to have at least one failing test that actually needs viewpoint adaptation.
Tests below have no RDQ, so viewpoint adaptation is boring. Also, they only test for incompatible arguments, not that the constructed object is actually what you want, e.g. have the right argument, but the constructed object from `super` is not what the current constructor expects.

Reply:
Thanks, I have expanded the test cases.